src/challenge/components/Outputs/SessionFiles.tsx

```diff
@@ -0,0 +1,188 @@
+    // accept: {
+    //   "application/zip": [".zip"]
+    // },
+    maxFiles: 1,
+    onDropAccepted: (files) => {
+      if (files && files.length === 1) {
+        propsRef.current?.onUpload(files[0]);
+      }
+    },
+  });
+
+  const propsRef = useRef<FileUploadControlProps | null>(null);
+  useEffect(() => {
+    propsRef.current = props;
+  }, [props]);
+
+  const theme = useTheme();
+
+  const focusedStyle = useMemo(() => {
+    return {
+      borderColor: theme.palette.primary.main,
+    };
+  }, [theme]);
+
+  const style = useMemo(
+    () => ({
+      ...baseStyle,
+      borderColor: theme.palette.secondary.main,
+      ...(isFocused ? focusedStyle : {}),
+      color: theme.palette.text.secondary,
+      backgroundColor: theme.palette.background.paper,
+    }),
+    [isFocused, focusedStyle, theme]
+  );
+
+  return (
+    <Box sx={{ maxWidth: 345, m: 1 }}>
+      <Box {...getRootProps({ style })}>
+        <input {...getInputProps()} />
+        {isDragActive ? (
+          <p>Drop the file here ...</p>
+        ) : (
+          <p>Drag &#38; drop or click to upload as a session file</p>
+        )}
+      </Box>
+    </Box>
+  );
+};
+
+const SessionFiles = React.forwardRef<SessionFilesHandle, SessionFilesProps>(
+  (props, ref) => {
+    // visual state
+    const [addFileDialogOpen, setAddFileDialogOpen] = useState<boolean>(false);
+
+    const propsRef = useRef<SessionFilesProps | null>(null);
+    useEffect(() => {
+      propsRef.current = props;
+    }, [props]);
+
+    return (
+      <>
+        <Card sx={{ height: "100%", overflowY: "auto" }}>
+          <CardContent sx={{ height: "100%" }}>
+            <Box sx={{ width: "300px", flexGrow: 1 }}>
+              <span style={{ fontFamily: "monospace" }}>session/</span>
+              <List
+                sx={{
+                  width: "100%",
+                  maxWidth: 360,
+                  bgcolor: "background.paper",
+                }}
+              >
+                {props.sessionFiles.map((file, index) => (
+                  <ListItem key={index} disablePadding>
+                    <IconButton
+                      onClick={() => {
+                        props.onRemoveSessionFile(file.filename);
+                      }}
+                    >
+                      <Tooltip title="Remove">
+                        <DeleteIcon color="error" />
+                      </Tooltip>
+                    </IconButton>
+
+                    <ListItemText primary={file.filename} />
+                  </ListItem>
+                ))}
+                <ListItem>
+                  <ListItemButton onClick={() => setAddFileDialogOpen(true)}>
+                    <ListItemText>+ new text file</ListItemText>
+                  </ListItemButton>
+                </ListItem>
+              </List>
+              <FileUploadControl
+                onUpload={(file) => {
+                  if (file.type.startsWith("text/")) {
```

Comment:
this doesn't always work, but worth trying to detect text files (so they show up in the tabs)